runtime-modules/content/src/lib.rs

```diff
@@ -1261,6 +944,548 @@ decl_module! {
+        ) {
+
+            // Ensure given video exists
+            let video = Self::ensure_video_exists(&video_id)?;
+
+            // Ensure there is no pending offer or existing auction for given nft.
+            video.ensure_nft_transactional_status_is_idle::<T>()?;
+
+            // Authorize nft owner
+            Self::authorize_nft_owner(
+                origin,
+                &actor,
+                &video
+            )?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Set nft transactional status to InitiatedOfferToMember
+            let video = video.set_pending_offer_transactional_status(to, price);
+
+            VideoById::<T>::insert(video_id, video);
+
+            // Trigger event
+            Self::deposit_event(RawEvent::OfferStarted(video_id, actor, to, price));
+        }
+
+        /// Cancel NFT offer
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn cancel_offer(
+            origin,
+            actor: ContentActor<CuratorGroupId<T>, CuratorId<T>, MemberId<T>>,
+            video_id: T::VideoId,
+        ) {
+
+            // Ensure given video exists
+            let video = Self::ensure_video_exists(&video_id)?;
+
+            // Ensure given pending offer exists
+            video.ensure_pending_offer_exists::<T>()?;
+
+            // Authorize nft owner
+            Self::authorize_nft_owner(
+                origin,
+                &actor,
+                &video
+            )?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Cancel pending offer
+            let video = video.set_idle_transactional_status();
+
+            VideoById::<T>::insert(video_id, video);
+
+            // Trigger event
+            Self::deposit_event(RawEvent::OfferCancelled(video_id, actor));
+        }
+
+        /// Accept incoming NFT offer
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn accept_incoming_offer(
+            origin,
+            video_id: T::VideoId,
+            participant_id: MemberId<T>,
+        ) {
+
+            // Authorize participant under given member id
+            let receiver_account_id = ensure_signed(origin)?;
+            ensure_member_auth_success::<T>(&participant_id, &receiver_account_id)?;
+
+            // Ensure given video exists
+            let video = Self::ensure_video_exists(&video_id)?;
+
+            // Ensure new pending offer is available to proceed
+            Self::ensure_new_pending_offer_available_to_proceed(&video, participant_id, &receiver_account_id)?;
+
+            if let Some(owned_nft) = &video.nft_status {
+
+                let owner_account_id = Self::ensure_owner_account_id(&video, &owned_nft)?;
+
+                //
+                // == MUTATION SAFE ==
+                //
+
+                // Complete nft offer
+                let video = Self::complete_nft_offer(video, owner_account_id, receiver_account_id);
```

Comment:
Very similar situation here, has to do lots of assumptions statically as if they don't hold, like as if there is no active initiated offer. This should be redone based on values provided from check above.

Reply:
Reworked here https://github.com/Joystream/joystream/pull/2512/commits/bc2e269a578d14bdba7cd92bf5b5715eab6ea0d3

Reply:
Great.